### DISCUSSION TITLE
dashboard 里面添加了域名和https 相关的文件，但是不生效

### DISCUSSION THREAD

**Discussion by thinkeng** [bfenetworks/bfe#960] [Q&A (问题和回答)] [answered]
**Describe the bug**
dashboard 里面添加了域名和https 相关的文件，但是不生效

wf 日志中

[2022/01/21 01:29:18 CST] [EROR] [17482] (github.com/baidu/go-lib/web-monitor/web_monitor.(*MonitorServer).reloadHandler:264) MonitorServer:Reload through web, cmd=[tls_conf], params=[127.0.0.1:42458], from[map[path:[/root/bfe/bfe/conf/tls_conf_20220121004408]]], err=[in CheckTlsConf() :certificate example.org not exist]

**Comment by mileszhang2016** [comment]
能把问题说的更详细一些吗？

**Comment by thinkeng** [comment, marked as answer]
我搭建的环境是，bfe  ；conf-agent  api-server，然后在后台的dashboard 里面系统管理下面，有个证书管理，我把对应域名的证书上传后，看bfe/conf 有tls_conf_时间 对应的目录，里面证书文件都加载过来了，但是里面的文件时儿有，时儿没有，然后将bfe的https 端口改为 HttpsPort=443，然后用域名访问的时候，加载下来的是   （签发者：example.org） 证书，不是对应域名的证书，看日志就是报上面那些错误

**Reply by liuqing6767** [reply]
对于tls配置，现在api-server不支持 配置 的文件有：
- client_ca
- client_crl
- session_ticket_key.data
- tls_rule_conf.data

能配置和下发的文件有：
- server_cert_conf.data  
 
在热加载tls配置时， tls_rule_conf.data 内容和 server_cert_conf.data 内容相关联。当两者不一致时就会出现关联检查失败而报错。

这个问题的解决方案是自行维护 tls_rule_conf.data 文件，按照业务需求手动修改该文件的原始内容。如果你不需要对租户进行自定义，修改方案为：

```
默认配置内容：
{
    "Version": "12",
    "DefaultNextProtos": ["http/1.1"],
    "Config": {
        "example_product": {
            "VipConf": [
                "10.199.4.14"
            ],
            "SniConf": ["example.org"],
            "CertName": "example.org",
            "NextProtos": [
                "h2;rate=100;isw=65535;mcs=200;level=0",
                "http/1.1"
            ],
            "Grade": "C",
            "ClientAuth": false,
            "ClientCAName": "example_ca"
        }
    }
}

修改后的简化配置：
{
    "Version": "12",
    "DefaultNextProtos": ["http/1.1"],
    "Config": {
    }
}

```

其他暂不支持动态加载的配置文件也按需进行手工维护即可。
